plugins/tiddlywiki/freelinks/text.js

```diff
@@ -3,160 +3,285 @@ title: $:/core/modules/widgets/text.js
 type: application/javascript
 module-type: widget
 
-An override of the core text widget that automatically linkifies the text
+An optimized override of the core text widget that automatically linkifies the text, with support for non-Latin languages like Chinese, prioritizing longer titles, skipping processed matches, excluding the current tiddler title from linking, and handling large title sets with enhanced Aho-Corasick algorithm.
 
 \*/
 
 "use strict";
 
 var TITLE_TARGET_FILTER = "$:/config/Freelinks/TargetFilter";
+var WORD_BOUNDARY_TIDDLER = "$:/config/Freelinks/WordBoundary";
 
 var Widget = require("$:/core/modules/widgets/widget.js").widget,
 	LinkWidget = require("$:/core/modules/widgets/link.js").link,
 	ButtonWidget = require("$:/core/modules/widgets/button.js").button,
-	ElementWidget = require("$:/core/modules/widgets/element.js").element;
+	ElementWidget = require("$:/core/modules/widgets/element.js").element,
+	AhoCorasick = require("$:/core/modules/utils/aho-corasick.js").AhoCorasick;
+
+var ESCAPE_REGEX = /[\\^$*+?.()|[\]{}]/g;
+
+function escapeRegExp(str) {
+	try {
+		return str.replace(ESCAPE_REGEX, '\\$&');
+	} catch(e) {
+		return null;
+	}
+}
+
+function FastPositionSet() {
+	this.set = new Set();
+}
+
+FastPositionSet.prototype.add = function(pos) {
+	this.set.add(pos);
+};
+
+FastPositionSet.prototype.has = function(pos) {
+	return this.set.has(pos);
+};
 
 var TextNodeWidget = function(parseTreeNode,options) {
 	this.initialise(parseTreeNode,options);
 };
 
-/*
-Inherit from the base widget class
-*/
 TextNodeWidget.prototype = new Widget();
 
-/*
-Render this widget into the DOM
-*/
 TextNodeWidget.prototype.render = function(parent,nextSibling) {
 	this.parentDomNode = parent;
 	this.computeAttributes();
 	this.execute();
 	this.renderChildren(parent,nextSibling);
 };
 
-/*
-Compute the internal state of the widget
-*/
 TextNodeWidget.prototype.execute = function() {
 	var self = this,
-		ignoreCase = self.getVariable("tv-freelinks-ignore-case",{defaultValue:"no"}).trim() === "yes";
-	// Get our parameters
+		ignoreCase = self.getVariable("tv-freelinks-ignore-case",{defaultValue:"no"}) === "yes";
```

Comment:
```suggestion
		ignoreCase = self.getVariable("tv-freelinks-ignore-case",{defaultValue:"no"}).trim() === "yes";
```

IMO we should keep the `trim()`, since we do handle end-user input